public/app/percona/add-instance/components/AddRemoteInstance/FormParts/MainDetails/MainDetails.tsx

```diff
@@ -45,18 +47,20 @@ export const MainDetailsFormPart: FC<MainDetailsFormPartProps> = ({ remoteInstan
         <LinkTooltip tooltipText={Messages.form.tooltips.mainDetails.username} icon="info-circle" />
       </div>
       <TextInputField
+        key={`username-${tlsFlag}`}
         name="username"
         placeholder={Messages.form.placeholders.mainDetails.username}
-        validators={[validators.required]}
+        validators={userPassValidators}
```

Comment:
just to be sure: is this working properly (i.e. checkbox state matches the correct validation by checking/unchecking)? Because I had problems with dynamic validators in the past.

Reply:
Yes it works, you have to change the key of the element for it to be rendered again and the validators to change. As you can see the key here depends on `tlsFlag`.